examples/server/utils.hpp

```diff
@@ -737,28 +737,50 @@ static json format_embeddings_response_oaicompat(const json & request, const jso
     return res;
 }
 
-static json format_response_rerank(const json & request, const json & ranks) {
-    json data = json::array();
-    int32_t n_tokens = 0;
-    int i = 0;
-    for (const auto & rank : ranks) {
-        data.push_back(json{
-            {"index",    i++},
-            {"relevance_score", json_value(rank, "score", 0.0)},
-        });
+static json format_response_rerank(
+        const json & request,
+        const json & ranks,
+        bool is_tei_format,
+        std::vector<std::string> & texts) {
+    json res;
+    if (is_tei_format) {
+        // TEI response format
+        res = json::array();
+        bool return_text = json_value(request, "return_text", false);
+        for (const auto & rank : ranks) {
+            int index = json_value(rank, "index", 0);
+            json elem = json{
+                {"index", index},
+                {"score", json_value(rank, "score", 0.0)},
+            };
+            if (return_text) {
+                elem["text"] = texts[index];
```

Comment:
Alternatively we can `std::move` here to avoid string copies.

Reply:
```suggestion
                elem["text"] = std::move(texts[index]);
```

Reply:
I mean that either we add `const` to the argument, or alternatively keep it non-const and use `std::move`. Applying both would still result in string copies.

Reply:
Oh ok right, I will remove the const then, while keeping std::move